bzt/jmx.py

```diff
@@ -648,14 +649,43 @@ def _get_cache_mgr():
         return mgr
 
     @staticmethod
-    def _get_cookie_mgr():
+    def _get_cookie_mgr(scenario):
         """
         :rtype: lxml.etree.Element
         """
         mgr = etree.Element("CookieManager", guiclass="CookiePanel", testclass="CookieManager", testname="Cookies")
         mgr.append(JMX._bool_prop("CookieManager.clearEachIteration", True))
         mgr.append(JMX._string_prop("CookieManager.implementation",
                                     "org.apache.jmeter.protocol.http.control.HC4CookieHandler"))
+        cookies = scenario.get(Scenario.COOKIES, [])
+        if cookies:
+            cookies_coll = JMX._collection_prop("CookieManager.cookies")
+            mgr.append(cookies_coll)
+            for cookie in cookies:
+                if not isinstance(cookie, dict):
+                    raise TaurusConfigError("Cookie must be dictionary: %s" % cookie)
+                c_name = cookie.get("name", TaurusConfigError("Name of cookie isn't found: %s" % cookie))
+                c_value = cookie.get("value", TaurusConfigError("Value of cookie isn't found: %s" % cookie))
+                c_domain = cookie.get("domain", TaurusConfigError("Domain of cookie isn't found: %s" % cookie))
+                c_path = cookie.get("path", "")
+                c_secure = cookie.get("secure", False)
+
+                # follow params are hardcoded in JMeter
```

Comment:
Let's investigate what happens in JMeter if c_path_specified and c_domain_specified are false. Maybe there is way to use real values for these flags in Taurus.